src/storage-client/src/client.rs

```diff
@@ -494,16 +514,27 @@ where
     T: timely::progress::Timestamp,
 {
     fn observe_command(&mut self, command: &StorageCommand<T>) {
+        // Note that `observe_command` is quite different in `mz_compute_client`.
+        // Compute (currently) only sends the command to 1 process,
+        // but storage fan's out to all workers, allowing the storage processes
+        // to self-coordinate how commands and internal commands are ordered.
+        //
+        // TODO(guswynn): cluster-unification: consolidate this with compute.
         match command {
+            StorageCommand::CreateTimely { .. } => {
+                // Similarly, we don't reset state here like compute, because,
+                // until we are required to manage multiple replicas, we can handle
+                // keeping track of state across restarts of storage server(s).
+            }
             StorageCommand::CreateSources(ingestions) => {
                 for ingestion in ingestions {
                     for &export_id in ingestion.description.source_exports.keys() {
                         let mut frontier = MutableAntichain::new();
-                        frontier.update_iter(iter::once((T::minimum(), i64::from(self.parts))));
-                        let part_frontiers = vec![
-                            Some(Antichain::from_elem(T::minimum()));
-                            usize::cast_from(self.parts)
-                        ];
+                        // TODO(benesch): fix this dangerous use of `as`.
```

Comment:
If you want to keep this it should probably be `TODO(gus)` :D

Reply:
its true!